tests/unit/test_dimension_utilities.py

```diff
@@ -304,6 +311,141 @@ def test_prefetch_dimension_variables(self, mock_get_opendap_nc4):
                                                      output_dir, self.logger,
                                                      access_token, self.config)
 
+        mock_add_bounds_variables.assert_called_once_with(prefetch_path,
+                                                          required_dimensions,
+                                                          self.varinfo, self.logger)
+
+
+    @patch('hoss.dimension_utilities.needs_bounds')
+    @patch('hoss.dimension_utilities.write_bounds')
+    def test_add_bounds_variables(self, mock_write_bounds, mock_needs_bounds):
+        """ Ensure that `write_bounds` is called when it's needed,
+            and that it's not called when it's not needed.
+
+        """
+        prefetch_dataset_name = 'tests/data/ATL16_prefetch.nc4'
+        varinfo_prefetch = VarInfoFromDmr(
+                'tests/data/ATL16_prefetch.dmr'
+        )
+        required_dimensions = {'/npolar_grid_lat','/npolar_grid_lon',
+                               '/spolar_grid_lat','/spolar_grid_lon',
+                               '/global_grid_lat','/global_grid_lon'}
+
+        with self.subTest('Bounds need to be written'):
+            mock_needs_bounds.return_value = True
+            add_bounds_variables(prefetch_dataset_name,
+                                 required_dimensions,
+                                 varinfo_prefetch,
+                                 self.logger)
+            self.assertEqual(mock_write_bounds.call_count, 6)
+
+            mock_needs_bounds.reset_mock()
+            mock_write_bounds.reset_mock()
+
+        with self.subTest('Bounds should not be written'):
+            mock_needs_bounds.return_value = False
+            add_bounds_variables(prefetch_dataset_name,
+                                 required_dimensions,
+                                 varinfo_prefetch,
+                                 self.logger)
+            mock_write_bounds.assert_not_called()
+
+
+    def test_needs_bounds(self):
```

Comment:
Kind of a nit: The line length used in all the service code tries to be 80 characters or less. The tests are a little laxer (because sometimes it's hard and/or makes things less readable), and so things around 85 or so characters slip in. But this test in particular seems to be stretching it a little (some lines are above 100 characters, which is one of the other arbitrary line lengths people opt for).

If this was all clocking in around 80 - 90 characters, it'd be a bit of a shrug from me (because this is a test file), but getting this long makes me think it would be nice to make this test code closer to the general aim of 80 characters where possible.

Reply:
I typically try to keep an eye on line length, but guess I zipped past them this go around.

Reply:
Thanks for pandering to my OCD!